spec/requests/requests_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.describe 'managing topics', :type => :request do
+
+  let!(:user) {FactoryBot.create :user}
+
+  it 'redirects unauthenticated users' do
+    post "/topics", :params => { :topic => {:name => "here is a new topic"} }
+    expect(response).to redirect_to('/users/sign_in')
+  end
+
+  it 'allows topic posting for logged in users' do
+    post user_session_path, :params => {'user[email]' => user.email, 'user[password]' => user.password }
+    get '/topics/new'
+    expect(response).to render_template(:new)
+
+    post "/topics", :params => { :topic => {:name => "here is a new topic"} }
+    expect(response).to redirect_to('/topics')
+    follow_redirect!
+
+    expect(response.body).to include("here is a new topic")
```

Comment:
Don't need this assertion. We just care about whether the response was a redirect to login.